connector_woocommerce/models/product_product/product_pricelist_item.py

```diff
@@ -0,0 +1,41 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class PricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    def _update_woocommerce_write_date_by_pricelist(self, values):
+        product = values.get("product_id", self.product_id)
+        if not isinstance(product, models.Model):
+            product = self.env["product.product"].browse(product).exists()
+        if not product:
+            product = values.get("product_tmpl_id", self.product_tmpl_id)
+            if not isinstance(product, models.Model):
+                product = self.env["product.template"].browse(product).exists()
+            product = product.product_variant_ids
+
+        if product.woocommerce_bind_ids:
+            product.woocommerce_write_date = fields.Datetime.now()
```

Comment:
Other option:
    def _update_woocommerce_write_date_by_pricelist(self, values):
        product = values.get("product_id")
        if product:
            product = self.env["product.product"].browse(product).exists()
        else:
            product = values.get("product_tmpl_id")
            if product:
                product = self.env["product.template"].browse(product).exists()
                product = product.product_variant_ids
        if not product:
            product = self.product_id or self.product_tmpl_id.product_variant_ids

        if product.woocommerce_bind_ids:
            product.woocommerce_write_date = fields.Datetime.now()